collect_subprojects.py

```diff
@@ -0,0 +1,59 @@
+#! /usr/bin/env python
+"""
+A script to generate a flat subproject directory from a tree of dependency directories
+"""
+import os
+import subprocess
+import textwrap
+import json
+import shutil
+import sys
+import warnings
+
+def project_name():
+    return os.path.split( os.getcwd() )[1]
+
+def dependency_directory():
+    """
+    A function isolating the path specification to the project's dependency directory
+    """
+    return os.path.join( os.getcwd(), "dependencies" )
+
+def traverse_dependencies( destination, traversed ):
+    """
+    collect unique links to dependency projects in a destination folder
+    """
+    if not os.path.isdir( dependency_directory() ):
+        return
+    
+    os.chdir( dependency_directory() )
```

Comment:
I always get nervous about using `chdir` because I can't keep track of what directory I'm in.